cmd/check_example2/main.go

```diff
@@ -0,0 +1,41 @@
+package main
+
+import (
+	"github.com/NETWAYS/go-check"
+	"github.com/NETWAYS/go-check/perfdata"
+	"github.com/NETWAYS/go-check/result"
+)
+
+func main() {
+	defer check.CatchPanic()
+
+	var overall result.Overall
+
+	check1 := result.PartialResult{}
+
+	check1.Output = "Check1"
+	check1.SetState(check.OK)
+	check1.Perfdata.Add(&perfdata.Perfdata{
+		Label: "foo",
+		Value: 23,
+	})
+
+	check2 := result.PartialResult{}
+
+	check2.Output = "Check2"
+	check2.SetState(check.Warning)
```

Comment:
SetState returns an error. I think even the examples should use proper coding style since it is what people copy-pasta.

Reply:
That is correct. Although I have mixed feelings about this (returning an error for `SetState`) since the only way you can get it, is to misuse this function with invalid input.